### ISSUE TITLE
Is diagram name an official feature?

### ISSUE THREAD

**Issue by dawngerpony** [qjebbs/vscode-plantuml#393]
When I load a diagram without a name, I see a warning:

```
Diagram unnamed. Try "@startuml name"
```

I see nothing in the [PlantUML specification][1] about naming diagrams, but I did find an issue (https://github.com/Enteee/plantuml-parser/issues/24) related to the subject, but with no clear conclusion.

**Should** a warning be generated when a diagram has no name, given that it doesn't seem to be an official PlantUML feature?

[1]: https://plantuml.com/sitemap-language-specification

**Comment by fuhrmanator** [comment]
I use the feature a lot, and I think it's a useful warning because the other features offered by the extension (generating markdown, exporting files) rely on having a name.

**Comment by qjebbs** [comment]
@dawngerpony of course it's an official feature, if not, why there is no parse error for this syntax? It's only not documented, but mentioned in some posts of the forum. But as @fuhrmanator said, this features is useful, so I chose to show those msg.

**Comment by dawngerpony** [comment]
Fair enough, thanks for your thoughts! I agree that it's a useful feature, I just wish it was documented on [plantuml.com](https://plantuml.com/) somewhere.

**Comment by Carglass** [comment]
Hello, I was wondering if there was a way to remove those warnings though? or an alternative to use the name of the file rather than the title to name the exports? I removed the titles so I could get the same filename by default when I export, but I am now flooded with these messages, and I cant quite figure out how to ignore them. I could not find an easy way to ignore this specific rule through VSCore but that may be my inexperience

**Comment by xpe** [comment]
qjebbs wrote:
> "of course it's an official feature"

@qjebbs Are you sure? How? Where is it documented?

I looked on these pages:
* https://plantuml.com/class-diagram
* https://plantuml.com/preprocessing

See also:
* https://forum.plantuml.net/10366/are-diagram-names-a-supported-feature

**Comment by The-Lum** [comment]
Hello all,

FYI: it is _quickly_ defined here:
- https://plantuml.com/sources

Regards.

**Comment by chipbite** [comment]
PlantUML gives users to set the diagram filename, as an **optional**. 
As a long time user of plantuml, I would recommend to avoid it and only use it when you have a special need - following the principle to keep it as simple as possible.

So, PlantUML does _not_ require the setting of the diagram name. 
To play nice with other tools (i e to ensure that behavior is consistent along the toolchain), neither should any tools based on it.

As for reference, there is no mention of the diagram filename setting in the [language reference guide](http://plantuml.com/guide), but PlantUML mentions it in the [PlantUML FAQ](https://plantuml.com/faq) when discussing the @startXXX tags, and clarifies that the filename is _optional_:

> you can optionally put a filename after the `@startXYZ`. This also allows to have several diagrams inside the same file.

![image](https://user-images.githubusercontent.com/559199/132870112-68f4130e-427f-4ab6-8d71-05059e7fdeb0.png)
